src/hotspot/share/memory/virtualspace.hpp

```diff
@@ -61,16 +62,16 @@ class ReservedSpace {
                           size_t page_size, bool special, bool executable);
 
   void initialize(size_t size, size_t alignment, size_t page_size,
-                  char* requested_address, bool executable);
+                  char* requested_address, bool executable, MemTag mem_tag = mtNone);
```

Comment:
When `mtNone` is default, it is still possible to introduce/create regions of Unknown types. The MemTag arg should be mandatory.

Reply:
Sorry Afshin, I didn't mention it in my summary at the top, but the goal in this fix is only to address the cases covered by the issue. I tried fixing all of them in my 1st try, but Thomas had some reservations. The "missing" cases will be addressed in a follow up(s).

However, I will copy all your feedback, so we don't loose it.

Thank you very much for taking detailed look and providing feedback!